### PR TITLE
reference new testserver

### PR DESCRIPTION
**Pull request by olydis**

### CODE REVIEW

**Comment by azuresdkci1x** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by azuresdkci1x** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

File: test/vanilla/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   "dependencies": {
     "@types/mocha": "^2.2.39",
     "@types/should": "^8.1.30",
-    "@types/node": "^6.0.66",
+    "@types/node": "7.0.0",
```
Comment (olydis) [inline test/vanilla/package.json]: @amarzavery FYI, more recent node typings make your code fail to compile (something `headers` were earlier typed as `any`, now as `string[] | string` AFAIK)

File: package.json [hunk 2]
```diff
@@ -45,7 +45,7 @@
     "shx": "^0.2.2",
     "through2-parallel": "^0.1.3",
     "yargs": "^8.0.2",
-    "@microsoft.azure/autorest.testserver": "1.9.2",
+    "@microsoft.azure/autorest.testserver": "2.3.9",
```
Comment (olydis) [inline package.json]: @amarzavery FYI, bumped targeted test server version, remaining files are just from the regen (except for the package.json that I also commented on 😉 )

Your unit tests seem to fail right now (independent of this, see https://github.com/Azure/autorest.nodejs/pull/13). If you fix them, you will get [this](https://github.com/Azure/autorest.java/pull/38#issuecomment-344767962) commented on a PR automatically 😉

**Pull request merged**